Packs/Expanse/Integrations/Expanse/Expanse.py

```diff
@@ -698,29 +696,56 @@ def fetch_incidents_command():
-            demisto.debug("Adding event incident name={name}, type={type}, severity={severity}".format(**incident))
-        incidents += event_incidents
-        if page_token is None:
-            more_events = False
-
-    # Fetch Behavior
-    if BEHAVIOR_ENABLED:
-        more_behavior = True
-        next_offset = None
-
-        while more_behavior:
-            behavior_incidents, next_offset = fetch_behavior_incidents_command(start_date, token, next_offset)
-            for incident in behavior_incidents:
-                demisto.debug("Adding behavior incident name={name}, type={type}, severity={severity}".format(**incident))
-            incidents += behavior_incidents
-            if next_offset is None:
-                more_behavior = False
-
-    # Send all Incidents
-    demisto.incidents(incidents)
+    # Check if we've stored any events in the integration cache
+    cache = demisto.getIntegrationContext()
+    stored_incidents = cache.get("incidents")
+
+    if stored_incidents is None:
+        while more_events:
+            event_incidents, page_token = fetch_events_incidents_command(start_date, end_date, token, page_token)
+            for incident in event_incidents:
+                demisto.debug("Adding event incident name={name}, type={type}, severity={severity}".format(**incident))
+            incidents += event_incidents
+            if page_token is None:
+                more_events = False
+
+        # Fetch Behavior
+        if BEHAVIOR_ENABLED:
+            more_behavior = True
+            next_offset = None
+
+            while more_behavior:
+                behavior_incidents, next_offset = fetch_behavior_incidents_command(start_date, token, next_offset)
+                for incident in behavior_incidents:
+                    demisto.debug("Adding behavior incident name={name}, type={type}, severity={severity}".format(**incident))
+                incidents += behavior_incidents
+                if next_offset is None:
+                    more_behavior = False
+
+        # Send PAGE_LIMIT number of incidents to demisto
+        if len(incidents) > PAGE_LIMIT:
+            incidents_to_send = incidents[:PAGE_LIMIT]
+            del incidents[:PAGE_LIMIT]
+            demisto.incidents(incidents_to_send)
+        else:
+            demisto.incidents(incidents)
+
+        # Add remaining incidents to cache
+        cache["incidents"] = incidents
+        demisto.setIntegrationContext(cache)
+    else:
+        # Send next PAGE_LIMIT number of incidents to demisto
+        if len(stored_incidents) > PAGE_LIMIT:
+            incidents_to_send = stored_incidents[:PAGE_LIMIT]
+            del stored_incidents[:PAGE_LIMIT]
+        else:
+            incidents_to_send = list(stored_incidents)
+            stored_incidents = None
+        demisto.incidents(incidents_to_send)
+
+        # Update Cache
+        cache["incidents"] = stored_incidents
+        demisto.setIntegrationContext(cache)
```

Comment:
i dont really like the change here
the cache isnt meant for that and it could introduce bugs in edge cases
why do we need that and not limit in the request as it was?

Reply:
Maybe we could loop @anara123 or @fvigo into the conversation here. We spoke about this in Slack - the real issue is with the Expanse API not being able to do any kind of limiting/filtering to enable us to respect the incident page limit param.

This approach include pulling all applicable events once, and storing them in the cache and popping off then publishing a portion of them during each incident fetch. The alternative approach would be similar, but we would only store ids in the cache, and during each `fetch-incidents` we would pull all events, and use the next X ids from the cache to determine what will get published as an incident.

I like this approach more because we will only hit the API once (less network traffic/latency) and because it's slightly less complex.

Reply:
I'm also more than happy to look into other options for accomplishing this - unfortunately the APIs we're using are unlikely to change to allow a more straight-forward approach to this.

Reply:
@andrew-expanse how big is the response returned from the API ? trying to understand if there will be an actual impact on the performance if we query for all incidents every fetch

Reply:
It will vary quite a bit by customer, but I'd anticipate it could be 5-10 requests to paginate through all of the data each time. Due to API limitations we're doing client-side filtering on severity, so a large majority of the data pulled is likely thrown away each time. 

From an earlier conversation about this I estimated that the max size we would be storing in the cache would be 
```
My guess on max size would be ~30MB assuming no filtering is done, but since we have a default filter in place I would expect things to be ~10MB on the high end, typically lower.
```
The response to this was 
```
Load balancing group should not have a problem.
10MB-30MB shouldnt be a problem.
But still worth to validate it.
But I would say 30MB is the limit
```